sdk/cosmos/azure-cosmos-spark_3_2-12/src/main/scala/com/azure/cosmos/implementation/SparkBridgeImplementationInternal.scala

```diff
@@ -192,6 +195,39 @@ private[cosmos] object SparkBridgeImplementationInternal extends BasicLoggingTra
     rangeToNormalizedRange(feedRange.getEffectiveRange(pkDefinition))
   }
 
+  private[cosmos] def partitionKeyValuesToNormalizedRange
+  (
+      partitionKeyValue: Object,
+      partitionKeyDefinitionJson: String
+  ): NormalizedRange = {
+
+      val partitionKey = new PartitionKeyBuilder()
+      val objectMapper = new ObjectMapper()
+      val json = partitionKeyValue.toString
+      var isMultiplePartitionKeyPaths = false
+      try {
+          val partitionKeyValues = objectMapper.readValue(json, classOf[Array[String]])
+          for (value <- partitionKeyValues) {
+              partitionKey.add(value.trim)
+          }
+          partitionKey.build()
+          isMultiplePartitionKeyPaths = true
+      } catch {
+          case e: Exception =>
+              logInfo("Invalid partition key paths: " + json, e)
+      }
+      if (!isMultiplePartitionKeyPaths) {
+          partitionKey.add(partitionKeyValue.toString.trim)
+      }
+
+      val feedRange = FeedRange
+          .forLogicalPartition(partitionKey.build())
+          .asInstanceOf[FeedRangePartitionKeyImpl]
+
+      val pkDefinition = SparkModelBridgeInternal.createPartitionKeyDefinitionFromJson(partitionKeyDefinitionJson)
+      rangeToNormalizedRange(feedRange.getEffectiveRange(pkDefinition))
```

Comment:
NIT - LF before line 228 to make it easier to grasp the return Value

Reply:
Done